src/foam/u2/ExportModal.js

```diff
@@ -121,15 +121,25 @@ foam.CLASS({
       var exportDriver = await this.exportDriverRegistryDAO.find(this.dataType);
       exportDriver = foam.lookup(exportDriver.driverName).create();
 
-      exportDriver.exportDAO(this.__context__, this.exportData)
-        .then(function(result) {
-          result = 'data:text/csv;charset=utf-8,' + result;
-          var link = document.createElement('a');
-          link.setAttribute('href', encodeURI(result));
-          link.setAttribute('download', 'data.csv');
-          document.body.appendChild(link);
-          link.click();
-        });
+      if ( this.exportData ) {
```

Comment:
Can you do something like this instead?
```
var p = this.exportData ?
    exportDriver.exportDAO(this.__context__, this.exportData) :
    Promise.resolve(exportDriver.exportFObject(this.__context__, this.exportObj));
p.then(result => {
            result = 'data:text/csv;charset=utf-8,' + result;
            var link = document.createElement('a');
            link.setAttribute('href', encodeURI(result));
            link.setAttribute('download', 'data.csv');
            document.body.appendChild(link);
            link.click();
})
```

Reply:
Done. Did the same for the above to convert to a ternary.